### PR TITLE
Moon 143 Cleric的工具类全部移到工具库，通过工具库引入

### PR DESCRIPTION
**Pull request by timsheng**

### CODE REVIEW

File: lib/lolitado/db.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    def query sql
+      result = db[sql].all
+      return result
+    end
+
+    def multiple_query sql
+      splited_sql = sql.split(';')
+      splited_sql.each do |each_sql|
+        query(each_sql)
+      end
```
Comment (dan-pan) [inline lib/lolitado/db.rb:29]: Does it return all result or the latest one?

File: lib/lolitado/db.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    end
+    
+    def query_duration sql, type = true, waiting_time = 10
+      start = Time.now
+      if type
+        result = query(sql, waiting_time)
```
Comment (dan-pan) [inline lib/lolitado/db.rb]: It should be query_wait instead of query. It's my mistake.

File: lib/lolitado.rb [hunk 3]
```diff
@@ -1,10 +1,26 @@
...
+    Graph.request(query, variables)
+  end
+
+  def add_headers value
+    API.add_headers value
+  end
+
+  def base_uri uri
+    API.base_uri uri
   end
```
Comment (dan-pan) [inline lib/lolitado.rb:80]: 这些是请求api的方法放在这边是否合适？

File: lib/lolitado/db.rb [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      msecs = (finish - start) * 1000.0
+      hash = {'result' => result, 'duration' => msecs}
+      return hash
+    end
+
+    ## Expect to return data for the query.
+    ## If the result is empty then it will be called again until waiting_time reach.
+    def query_wait sql, waiting_time = 10
+      result = db[sql].all
+      if result.length == 0
```
Comment (amber-yuan) [inline lib/lolitado/db.rb]: I'm not sure the class of "result" has to be Array or Hash,  if so, use "result.empty?" or "result.length.zero?" maybe  more ruby.

File: lib/lolitado/db.rb [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      db[sql]
+    end
+
+    def insert_by_data data, table
+      sql = "insert into #{table} "
+      case data
+        when Array
+          data.each do |d|
+            insert_by_data(d, table)
+          end
```
Comment (amber-yuan) [inline lib/lolitado/db.rb:107]: Why is there no query in this step，I'm confused.

File: lib/lolitado/db.rb [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+          sql = "#{sql} #{k}=#{v},"
+        else
+          sql = "#{sql} #{k}='#{v}',"
+        end
+      end
+      sql = sql[0..-2] + " where"
+      condition.each do |k,v|
+        sql = "#{sql} #{k} = '#{v}' and"
+      end
+      query(sql[0..-4],0)
```
Comment (dan-pan) [inline lib/lolitado/db.rb]: The query method is changed, it should be query(sql[0..-4])

File: lib/lolitado/db.rb [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+      case data
+        when Array
+          data.each do |d|
+            insert_by_data(d, table)
+          end
+        when Hash
+          columns = data.keys.to_s.gsub('[','(').gsub(']',')').gsub('"','')
+          values = data.values.to_s.gsub('[','(').gsub(']',')').gsub('nil','NULL')
+          sql = sql + columns + " values " + values
+          query(sql,0)
```
Comment (dan-pan) [inline lib/lolitado/db.rb]: The query method is changed, it should be query(sql)

File: lib/lolitado/db.rb (hunk 5 above)
Comment (dan-pan) [reply lib/lolitado/db.rb:107]: It's in insert_by_data method.

File: lib/lolitado/db.rb (hunk 4 above)
Comment (dan-pan) [reply lib/lolitado/db.rb]: The result have to be Array.
For yes, result.empty? is fine, it could be more ruby.

File: lib/lolitado/db.rb [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+      end
+    end
+
+    def update data, table, condition
+      sql = "update #{table} set"
+      data.each do |k,v|
+        v = v.to_json if v.is_a?(Hash)
+        if !!v == v
```
Comment (amber-yuan) [inline lib/lolitado/db.rb:127]: ...emm, What is the meaning of this code?

Comment (dan-pan) [reply lib/lolitado/db.rb:127]: 这行代码是判断是否是布尔值.

File: lib/lolitado.rb (hunk 3 above)
Comment (timsheng) [reply lib/lolitado.rb:80]: 这边只是做了个代理，将cleric里面需要用到的一些调httparty的方法转发过去

File: lib/lolitado/db.rb (hunk 1 above)
Comment (timsheng) [reply lib/lolitado/db.rb:29]: 我没有改过这些方法的实现，看代码逻辑的话应该是 all results

File: lib/lolitado/db.rb (hunk 2 above)
Comment (timsheng) [reply lib/lolitado/db.rb]: ok， i will correct it.

File: lib/lolitado/db.rb (hunk 4 above)
Comment (timsheng) [reply lib/lolitado/db.rb]: i will correct it

**Comment by timsheng** [comment]
等dan的commit去fix这些问题

**Comment by dan-pan** [comment]
我这边好了

**Pull request merged**